osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/InstallationService.java

```diff
@@ -122,12 +120,55 @@ private List<SecurityKeyType> determineKeyTypesToStore(final SmartMeteringDevice
             ComponentType.PROTOCOL_DLMS,
             new IllegalArgumentException(msg));
       }
-      return Arrays.asList(G_METER_MASTER);
+      return gMeterKeyTypes;
```

Comment:
Should you not always all key types as done for e-meter

Reply:
right, all key types are returned now.